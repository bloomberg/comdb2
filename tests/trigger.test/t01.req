CREATE TABLE t1 (x INT);$$

CREATE PROCEDURE trigger_1 VERSION 'trigger_1' {
  local function main(event)
    local tbl = db:table("tbl", {{"i", "int"}})
    local tp = event.type
    local inew, iold
    if tp == 'add' then
      inew = event.new.i
      tbl:insert({i=inew})
    elseif tp == 'del' then
      iold = event.old.i
      tbl:insert({i=iold})
    end
    return 0
  end
}$$

CREATE LUA TRIGGER trigger_1 ON (TABLE t1 FOR INSERT AND UPDATE AND DELETE)

BEGIN;
INSERT INTO t1 VALUES(1);
INSERT INTO t1 VALUES(2);
INSERT INTO t1 VALUES(3);
COMMIT;

BEGIN;
UPDATE t1 SET x = x + 1 WHERE x < 2;
COMMIT;

BEGIN;
UPDATE t1 SET x = x + 1 WHERE x > 3;
COMMIT;

BEGIN;
UPDATE t1 SET x = x * 9 WHERE x = 1;
COMMIT;

BEGIN;
DELETE FROM t1 WHERE x = 1;
COMMIT;

BEGIN;
DELETE FROM t1 WHERE x = 9;
COMMIT;

BEGIN;
DELETE FROM t1;
COMMIT;

BEGIN;
INSERT INTO t1 VALUES(4);
INSERT INTO t1 VALUES(5);
INSERT INTO t1 VALUES(6);
COMMIT;

SELECT x FROM t1 ORDER BY x;

DROP LUA TRIGGER trigger_1
