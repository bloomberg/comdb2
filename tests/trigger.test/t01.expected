(version='trigger_1')
(x=4)
(x=5)
(x=6)
